tools/dnulist/program.cs

```diff
@@ -0,0 +1,241 @@
+            
+        if (targetFound && packages.Count > 0)
+        {
+            foreach (var package in packages.Keys)
+            {   
+                Console.WriteLine("Copying package: {0}", package);
+                
+                var p = packages[package];         
+                if (p.Compile != null)
+                {
+                    foreach (var lib in p.Compile.Keys)
+                    {
+                        var source = Path.Combine(packageRoot,package,lib);
+                        var destination = Path.Combine(compilePath,Path.GetFileName(lib));;
+                        File.Copy(source,destination, true);
+                        
+                        if (verbose)
+                        {
+                            Console.WriteLine("**Compile: {0}", lib);
+                        }
+                    }
+                }
+                
+                if (p.Runtime != null)
+                {
+                    foreach (var lib in p.Runtime.Keys)
+                    {
+                        var source = Path.Combine(packageRoot,package,lib);
+                        var destination = Path.Combine(runtimePath,Path.GetFileName(lib));;
+                        File.Copy(source,destination,true);
+                        
+                        if (verbose)
+                        {
+                            Console.WriteLine("**Runtime: {0}", lib);
+                        }
+                    }
+                }
+            }
+        }
+        else if (targetFound)
+        {
+            Console.WriteLine("No packages found for {0}.",target);
+        }
+        else
+        {
+            Console.WriteLine("{0} is not a target for this project.", target);
+        }
+    }
+    
+    private static void PrintHelp()
+    {
+        Console.WriteLine("Primary arguments:");
+        Console.WriteLine("--target - prints targets");
+        Console.WriteLine("--target [target] - selects targets to print (the default) or otherwise act upon");
+        Console.WriteLine("--help - prints help");
+        Console.WriteLine();
+        Console.WriteLine("Additional arguments:");
+        Console.WriteLine("--copy [path to copy to] - copies reference and runtime assemblies for a target");
+        Console.WriteLine("--verbose - print maximum amount of info");
+    }
+    
+    private static Dictionary<string,string> GetArguments(string[] args)
+    {
+        const string ArgSyntax = "--";
+        var commands = new Dictionary<string,string>();
+        int count = 0;
+
+        while (count < args.Length)
+        {   
+            if (count +1 == args.Length && args[count].StartsWith(ArgSyntax))
+            {
+                commands.Add(args[count],string.Empty);
+            }
+            else if (count +1 == args.Length && !args[count].StartsWith(ArgSyntax))
+            {
+                // bad argument
```

Comment:
Perhaps you could use the unqualified syntax to represent the file to operate on?  I think in that case you could turn this into  if (!args[count].StartsWith(ArgSyntax))